unit-tests/src/test/java/org/eclipse/collections/impl/list/IntervalTest.java

```diff
@@ -77,6 +83,20 @@ public void zeroToBy_throws_step_size_zero()
         Interval.zeroToBy(0, 0);
     }
 
+    @Test
+    public void fromToBy_throws_on_oversized_step()
+    {
+        Verify.assertThrows(
+                IllegalArgumentException.class,
+                () -> Interval.fromToBy(1, 10, 10));
+        Verify.assertThrows(
+                IllegalArgumentException.class,
+                () -> Interval.fromToBy(1, -10, -12));
+        Verify.assertThrows(
+                IllegalArgumentException.class,
+                () -> Interval.fromToBy(1_000_000_000, 2_000_000_000, 1_000_000_001));
```

Comment:
There needs to be test case for this:
Verify.assertThrows(
                IllegalArgumentException.class,
                () -> IntInterval.fromToBy(1_000_000_000, 2_000_000_000, 1_500_000_000));